### PR TITLE
[Alignment Configuration] Rename align_cfg::alignment_result_capture to align_cfg::detail::result_type

### PR DESCRIPTION
**Pull request by simonsasse**
Resolves seqan/product_backlog#183

This moves the alignment_result_capture to a new namespace seqan3::align_cfg::detail and then renames it to result_type.
I needed to specify the whole namespace path seqan3::detail::XX for most of the view files. This is due to the newly created seqan3::align_cfg::detail which had a name clash with seqan3::detail.

Merging this PR closes #2018 because it has the same intention.

Blocked by #2033

### CODE REVIEW

File: include/seqan3/alignment/configuration/detail.hpp
Comment (joergi-w) [inline include/seqan3/alignment/configuration/detail.hpp:35]: This is different from the other lines... I don't know what's the impact, but please consider to keep the existing style.
```suggestion
    result_type,  //!< ID for the \ref seqan3::align_cfg::detail::result_type "result_type" option.
```

Comment (joergi-w) [reply include/seqan3/alignment/configuration/detail.hpp:35]: Attention: Here is `detail` code documentation in the user visible documentation!
@ second reviewer: Is it fine or should we avoid this? (how?)

File: CHANGELOG.md
Comment (joergi-w) [inline CHANGELOG.md:87]: Maybe also add the namespace for `alignment_result_type`?

**Comment by joergi-w** [comment]
Merging this PR closes #2018 because it has the same intention.

File: CHANGELOG.md
Comment (simonsasse) [reply CHANGELOG.md]: @rrahn correctly said that this is just developer API and thus does not need a changeling entry. Changed that now!

File: include/seqan3/alignment/configuration/detail.hpp
Comment (joergi-w) [inline include/seqan3/alignment/configuration/detail.hpp:35]: This produces an error in the user documentation build, because it does not know about detail members. Try to work with `\if DEV`, but could be difficult in a table...

Comment (marehr) [reply include/seqan3/alignment/configuration/detail.hpp:35]: all of these PR's have the problem, I added a working fix in https://github.com/seqan/seqan3/pull/2020

But seeing that multiple PRs need that, I'll extract that commit in a new PR.

**Comment by marehr** [comment]
@seqan/core We have now some redundant work #2020 they both worked around the ambiguous `seqan3::detail`, `seqan3::align_cfg::detail` namespace.

**Comment by rrahn** [comment]
At the end I will have a final sweep to check for consistency issues that might happen between the PRs.

**Comment by rrahn** [comment]
@simonsasse I am waiting on the rebase on #2020 before I review. So I can focus on the actual changes.

File: include/seqan3/alignment/configuration/detail.hpp
Comment (simonsasse) [reply include/seqan3/alignment/configuration/detail.hpp:35]: So I leave it as is ?

File: include/seqan3/alignment/configuration/align_config_result_type.hpp
Comment (rrahn) [inline include/seqan3/alignment/configuration/align_config_result_type.hpp:39]:
```suggestion
struct result_type_tag :
```
This should be renamed with the suffix _tag

File: include/seqan3/alignment/configuration/align_config_result_type.hpp
Comment (rrahn) [inline include/seqan3/alignment/configuration/align_config_result_type.hpp:42]:
```suggestion
    public pipeable_config_element<result_type_tag<alignment_result_t>,
                                   std::type_identity<alignment_result_t>>
{
```

File: include/seqan3/alignment/configuration/align_config_result_type.hpp
Comment (rrahn) [inline include/seqan3/alignment/configuration/align_config_result_type.hpp:75]:
```suggestion
inline constexpr seqan3::detail::result_type_tag<alignment_result_t> result_type{};
```

File: include/seqan3/alignment/configuration/detail.hpp
Comment (rrahn) [reply include/seqan3/alignment/configuration/detail.hpp:35]: yes

File: include/seqan3/alignment/pairwise/detail/type_traits.hpp
Comment (rrahn) [inline include/seqan3/alignment/pairwise/detail/type_traits.hpp:105]:
```suggestion
        if constexpr (configuration_t::template exists<result_type_tag>())
```

File: test/unit/alignment/configuration/align_config_common_test.cpp
Comment (rrahn) [inline test/unit/alignment/configuration/align_config_common_test.cpp:31]:
```suggestion
                                    seqan3::detail::result_type_tag<alignment_result_t>>;
```

**Review by rrahn — CHANGES_REQUESTED** [review]
Some general renaming, otherwise looks very good. Thank you

**Comment by simonsasse** [comment]
Thanks for the review @rrahn . I applied all the changes and also rebased on the latest version of the release branch because there where merge conflicts. On my machine tests pass 💻✅

File: include/seqan3/alignment/configuration/align_config_vectorised.hpp
Comment (marehr) [inline include/seqan3/alignment/configuration/align_config_vectorised.hpp:29]:
```suggestion
    static constexpr detail::align_config_id id{detail::align_config_id::vectorised};
```

**Review by marehr — APPROVED** [review]

**Comment by smehringer** [comment]
whats the state here?

**Comment by simonsasse** [comment]
If @marehr review counts as second review, its time for a merge I think :)

**Comment by smehringer** [comment]
@marehr anything to add? Otherwise I'll merge (or you if you see this).

**Comment by marehr** [comment]
I only waited on CI and forgot that I waited in CI :see_no_evil: (I'm not accustomed to be second reviewer) xD

**Pull request merged**
